homeassistant/components/mqtt/models.py

```diff
@@ -289,6 +289,30 @@ def async_render_with_possible_json_value(
         return rendered_payload
 
 
+class EntityMonitor:
+    """Monitors entity state changes."""
+
+    _singleton: object = object()
+
+    def __init__(self, entity: Entity, attributes: set[str]) -> None:
+        """Initialize entity monitor."""
+        self._entity: Entity = entity
+        self._attributes: dict[str, Any] = {
+            attribute: getattr(entity, attribute, self._singleton)
+            for attribute in attributes
+        }
+
+    @property
+    def changed(self) -> bool:
+        """Return True if attributes on entity changed or if update is forced."""
+        if getattr(self._entity, "_attr_force_update", False):
+            return True
+        return any(
+            getattr(self._entity, attribute, self._singleton) != last_value
+            for attribute, last_value in self._attributes.items()
+        )
```

Comment:
I'd probably write this out since python will have to generate an expression for the `any` which isn't inlined currently